### ISSUE TITLE
Regression on jump to definition in a comment reference for an inherited member

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#39467]
If I have a doc comment with `[Directory.watch]`. I can no longer jump to definition for `watch`. This is while using the analysis server with `--lsp` but I suspect the regression will surface in the normal mode as well.

This worked in Dart 2.5.0. This is broken in 2.6.0

Dart 2.5.0 behavior:

- Jumping to definition on `Directory` takes you to `abstract class Directory` in `directory.dart.
- Jumping to definition on `watch` in comment reference takes you to `watch` in `file_system_entry.dart`.
- Jumping to definition `watch` in non-comment usage takes you to the same place.
- There are no lints for `comment_references`.

Dart 2.6.0 behavior:

- Jumping to definition on `Directory` - same.
- Jumping to definition on `watch` in comment reference returns no locations.
- Jumping to definition on `watch` in non-comment usage takes you to the right place.
- `[Directory.watch]` is reported as a violation of the lint `comment_references`

**Comment by DanTup** [comment]
I'm not sure why this worked before and became broken, but by coincidence (my finding this issue today) I recently opened a fix for this because it also came up in https://github.com/Dart-Code/Dart-Code/issues/4114.

![Aug-25-2022 15-06-43](https://user-images.githubusercontent.com/1078012/186686972-43301d0e-7244-4eb6-a7d7-f1e926e127dc.gif)

I've also confirmed that the lint diagnostic disappears with the same fix.

I'll close this once https://dart-review.googlesource.com/c/sdk/+/256162 lands.
